### PR TITLE
Android RN distribution + back button fix

### PR DESCRIPTION
**Pull request by ferrannp**
### Android
- [x] No need for any of the Gitlab repos used before `react-native-hotels` and `react-native-native-modules`. No need to deploy to NPM anymore
- [x] Create playground app that can be used with the packager too (`yarn start`)

### JS
- [x] Fix native back button: Remove `WithBackButtonListener` HOC in favour of native module solution (like iOS) + both platforms together.
- [x] Fix back button for Toolbar when going back to native screen

You can also check the README: https://github.com/kiwicom/mobile/blob/android-rn-distribution/RNAndroidPlayground/README.md

### Android
* More automation for distribution (pushing SNAPSHOT(s))
* Take versions (like `react-native`, `react-native-maps`, etc) from `package.json`
* Make CircleCI ensures that `rnkiwimobile` and `RNAndroidPlayground` are building correctly

### JS
* Write a test that ensures exact versions for the dependencies we use

### iOS
* When this PR is merged, iOS needs to implement the toolbar back button function. Probably here https://github.com/kiwicom/mobile/pull/698.

### CODE REVIEW

File: RNAndroidPlayground/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        sourceCompatibility 1.8
+        targetCompatibility 1.8
+    }
+}
+
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+
+    // Uncomment next line to develop locally & run "yarn start"
+    // implementation project(':rnkiwimobile')
```
Comment (tbergq) [inline RNAndroidPlayground/app/build.gradle]: Does this mean I have to uncomment this line whenever I run locally, and remember to not commit it?

Comment (ferrannp) [reply RNAndroidPlayground/app/build.gradle]: I plan to change this in subsequents PRs. You still can run it locally (the way it is). I just didn't want to add more work into this PR. But for now if you want to run it with `implementation project` that's the way to go. Commiting it by mistake would not break anything though.

Comment (tbergq) [reply RNAndroidPlayground/app/build.gradle]: > I plan to change this in subsequents PRs

That is good enough for me.

File: RNAndroidPlayground/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,55 @@
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
+apply from: "../dependencies.gradle"
+
+project.ext.vectoricons = [
+        iconFontNames: ['MaterialIcons.ttf']
+]
+
+apply from: "../../node_modules/react-native-vector-icons/fonts.gradle"
```
[the remark this answers is not in the document]
Comment (ferrannp) [reply RNAndroidPlayground/app/build.gradle]: I am going to remove it, the `sh` should do it already.

**Pull request merged**